### DISCUSSION TITLE
API to Automate Rotation of Private Keys for Github App

### DISCUSSION THREAD

**Discussion by ahernandez411** [community/community#39274] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

I have a Github App that we are using in one of our repositories. We would like to automate the rotation of private keys for this App. I have been searching the API docs and I thought [Deploy Keys](https://docs.github.com/en/rest/deploy-keys#about-the-deploy-keys-api) might be what I use but I haven't been able to even perform list successfully.

When clicking the `Generate a private key` button on the app settings page I see, in the dev tool, that it first runs a post and then a get request to the following URL: `https://github.com/organizations/[orgName]/settings/apps/[apName]/key`. I'm trying to see if there is an equivalent API for this. 

We want to have the ability to run a Github Action and have it rotate this private key out and replace it with a new one. Is this possible using the GIthub API?

**Comment by official-mywebprojects** [comment]
I'll suggest you contact any of the developer teams in the Github developer member program for support. Try git@vger.kernel.org

**Reply by ahernandez411** [reply]
I'll give it a try. Thank you

**Comment by ahernandez411** [comment, marked as answer]
I got a response from Github Support; no such API currently exists to perform such an action.
